src/main/java/io/cdap/plugin/servicenow/source/ServiceNowBaseSourceConfig.java

```diff
@@ -163,39 +163,31 @@ private void validateValueType(FailureCollector collector) {
 
   private void validateDateRange(FailureCollector collector) {
     if (containsMacro(ServiceNowConstants.PROPERTY_START_DATE) ||
-      containsMacro(ServiceNowConstants.PROPERTY_END_DATE)) {
+      containsMacro(ServiceNowConstants.PROPERTY_END_DATE) ||
+      (Util.isNullOrEmpty(startDate) && Util.isNullOrEmpty(endDate))) {
       return;
     }
 
-    if (!Util.isNullOrEmpty(startDate) && !Util.isNullOrEmpty(endDate) &&
-      !Util.isValidDateFormat(ServiceNowConstants.DATE_FORMAT, startDate) &&
-      !Util.isValidDateFormat(ServiceNowConstants.DATE_FORMAT, endDate)) {
-      collector.addFailure("Invalid format for Start date. Correct Format: " +
-          ServiceNowConstants.DATE_FORMAT, null)
-        .withConfigProperty(ServiceNowConstants.PROPERTY_START_DATE);
-      collector.addFailure("Invalid format for End date. Correct Format:" +
-          ServiceNowConstants.DATE_FORMAT, null)
+    if (Util.isNullOrEmpty(startDate) || Util.isNullOrEmpty(endDate)) {
+      collector.addFailure("Enter values for both Start date and End date.", null)
+        .withConfigProperty(ServiceNowConstants.PROPERTY_START_DATE)
```

Comment:
this should only mark the problematic config property instead of both. There should be separate checks for start and end dates.

Reply:
this also returns early. Even if the start date is missing, the end date format should still be checked. Can move this check below the format checks.

Reply:
> this should only mark the problematic config property instead of both. There should be separate checks for start and end dates.

Added separate checks for start date and end date validation. b450d01ecde6caac6e9e789958aff4c54b3e4cb0